src/Symfony/Component/Form/ChoiceList/Factory/CachingFactoryDecorator.php

```diff
@@ -84,10 +84,8 @@ public function getDecoratedFactory()
     /**
      * {@inheritdoc}
      *
-     * @param callable|Cache\ChoiceValue|null  $value  The callable or static option for
-     *                                                 generating the choice values
-     * @param callable|Cache\ChoiceFilter|null $filter The callable or static option for
-     *                                                 filtering the choices
+     * @param mixed $value
```

Comment:
I don't think that this is really correct as we are forwarding the input and there is no guarantee that the decorated factory handles every input type as well.

Reply:
A decorator cannot provide this guarantee, but it cannot restrict the accepted types either.
The decorator will throw as expected if an incompatible type is provided.
That's the correct behavior IMHO.